packages/salesforcedx-vscode-soql/src/commands/soqlFileCreate.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { EDITOR_VIEW_TYPE } from '../constants';
+import { telemetryService } from '../telemetry';
+
+export async function soqlOpenNew() {
+  telemetryService.sendCommandEvent('soql_builder_open_new', process.hrtime());
+
+  if (vscode.workspace) {
+    // create untitled file
+    const newFileUri = vscode.Uri.file(
+      path.join(vscode.workspace.rootPath!, 'untitled.soql')
+    ).with({ scheme: 'untitled' });
+    await vscode.workspace.openTextDocument(newFileUri);
```

Comment:
I can't remember what this does exactly, it returns a text document right? IIRC opening a doc was a two step process or something. They separate "opening" a text document with viewing it in the editor or something like that?

Does this open the text editor ` vscode.workspace.openTextDocument(newFileUri);` 
then we quickly run ` vscode.commands.executeCommand(
      'vscode.openWith',
      newFileUri,
      EDITOR_VIEW_TYPE
    );` to switch it to the builder UI?

Reply:
`openTextDocument` enables a `TextDocument` object to be manipulated. It is the model for a file, the same model the an editor acts upon, but the model can exist independent of an editor as well. So that instruction does not open an editor.

Reply:
yeah, thats generally what I thought. I remember finding the name method `openTextDocument` to be misleading.